src/ui/mainWindow.ts

```diff
@@ -375,6 +375,16 @@ export const mainWindow = window({
 								value: model._z,
 								format: model._formatPosition,
 								onChange: (_, incr) => model._modifyVehicle(setPositionZ, incr)
+							}),
+							positionSpinner({
+								_label: { text: "Seat spin:" },
+								minimum: 0,
+								maximum: 255,
+								disabled: model._isPositionDisabled,
+								step: model._multiplier,
+								value: model._spin,
+								format: model._formatPosition,
+								onChange: (_, incr) => model._modifyVehicle(setSpin, incr)
```

Comment:
If I'm not mistaken, this could just be a `labelSpinner` instead of a `positionSpinner` as it doesn't need all the extra checks, so it can be implemented more like seats or mass (no format, no `incr` in `onChange`, include the missing `wrapMode`)

Also, there should probably be a different disabled store, one that checks if the vehicle can actually spin (similar to `_isUnpowered`).

Reply:
kinda hard to say when a vehicle is allowed to spin. there's some object flag that marks that the car rotates during turns but you could turn that off and still be able modify spinning cars to render other (spin) frames. Spin-less cars obviously just ignore that field.

Reply:
Wouldn't the flag only be incorrectly off on incorrectly configured CSO vehicles? I'd assume it would need some flag to be aware that there are spinning frames, but it's been a while since I looked at that code. 😅

Reply:
Well but even with the object turn spin flag turned off modifying the current spin frame is still useful.
![Clip-optimized](https://github.com/user-attachments/assets/cc2bb7c2-75cf-4b0b-ad60-96a262ea39fd)

Reply:
actually you could've done that with the spin locking track piece as well. (that toggles `VehicleFlags::SpinningIsLocked`). But cars with the turn-spin flag (`CAR_ENTRY_FLAG_SPINNING`) turned can still render spin sprites if we modify the car `spin` value.

Thing is, setting `VehicleFlags::SpinningIsLocked` will still reset the car spin when reaching the station (which is not desired for rides like the one in the gif). so turning off CAR_ENTRY_FLAG_SPINNING at ride object level + modifying the car spin value can enable some new tricks.

Reply:
so in short `CAR_ENTRY_FLAG_SPINNING` doesn't indicate if the car respect `Car.spin` field.

I see this bit in `src/openrct2/paint/vehicle/VehiclePaint.cpp`:
```
    if (carEntry->flags & CAR_ENTRY_FLAG_SPINNING_ADDITIONAL_FRAMES)
    {
        baseImageId += (vehicle->spin_sprite / 8) & 31;
    }
```

The only rides that use `spin_sprite` (`Car.spin`) without `CAR_ENTRY_FLAG_SPINNING_ADDITIONAL_FRAMES` are:
* `src/openrct2/paint/vehicle/Vehicle.RiverRapids.cpp`
* `src/openrct2/paint/vehicle/Vehicle.VirginaReel.cpp`

So maybe make the spinner disable test "!CAR_ENTRY_FLAG_SPINNING_ADDITIONAL_FRAMES && type!=rapids && type != virginiaReel`?

Reply:
`CAR_ENTRY_FLAG_SPINNING` is the flag I meant yes.

> But cars with the turn-spin flag (CAR_ENTRY_FLAG_SPINNING) turned off can still render spin sprites if we modify the car spin value.

But wouldn't this only occur on custom objects that don't have the correct flags? Or are there also built-in vehicles that can spin but don't have this flag?

Reply:
Don't think there are, but this is a tool meant for game state hacking anyway I presumed?

Reply:
It is a tool for game state hacking, but it can be overwhelming if there are too many options that don't do anything It's also consistent with how the powered spinners are disabled when they are of no use.